### ISSUE TITLE
Baseline correction

### ISSUE THREAD

**Issue by lpatiny** [image-js/image-js-typescript#275]
Often microscopy images have a problem with the baseline and I wonder how we could easily fix it.

I'm thinking about a gaussian filter followed by canny edge with a clear border or something like that in order to detect all the surfaces that are connected to the border.

And then sampling somehow the background and make a fit using a function. ([Paraboloid](https://en.wikipedia.org/wiki/Paraboloid) ?)

Here is a non confidential image that could be use to test. Maybe it can easily be done with the current version and it is a simple application that could move in the documentation.

![good](https://user-images.githubusercontent.com/1484241/230101422-ef27edc7-0025-42f6-a684-5e67a9ec914b.jpg)

**Comment by opatiny** [comment]
I have tried running the matlab example on the above image and get a black image with a radius 12 stucturing element. However I did obtain a lot better results by increasing the radius to 50 (result below). I have also observed that the bigger the structuring element, the slower the algorithm.
![screws-tophat-disk50](https://user-images.githubusercontent.com/17711389/232300268-65d26b76-8539-4aa3-8839-198eb6c004a8.png)

I also tried out with a 101x101 square structural element:
![screws-tophat-rectangle-101-101](https://user-images.githubusercontent.com/17711389/232300849-3e872c46-0e38-4b46-a99e-f03d566f98e3.png)

**Comment by opatiny** [comment]
Here is what I obtained by running the following script in `image-js`. I have tried a circular kernel which seemed very slow. In the end I am using a 101x101 square kernel. Running time is 5-10s. The result is very similar to matlab.

https://github.com/image-js/image-js-typescript/blob/da23765f76f914df1428aaa0e757ba8489e66847/scripts/tophatTest.ts

![background-topHat](https://user-images.githubusercontent.com/17711389/232307614-25bff7cd-a998-4bd8-b47f-bb00cd223293.png)
